hw_13/HW_13.py

```diff
@@ -0,0 +1,378 @@
+# 2. Функцию `long_lasting_function` из примера по понятным причинам запускать не нужно. Достаточно просто убедится, что большие временные интервалы правильно форматируются при отправке сообщения (как в примерах)
+# 3. Допустима реализация логирования, когда логгер полностью перехватывает запись в `stdout` и `stderr` (то есть при выполнении функций печать происходит **только** в файл)
+# 4. В реальной жизни вам не нужно использовать Telegram API при помощи ручных запросов, вместо этого стоит всегда использовать специальные библиотеки Python, реализующие Telegram API, они более высокоуровневые и удобные. В данном задании мы просто учимся работать с API при помощи написания велосипеда.
+# 5. Обязательно прочтите часть конспекта лекции про API перед выполнением задания, так как мы довольно поверхностно затронули это на лекции
+# 
+# **Рекомендуемые к использованию модули:**
+# 1. os
+# 2. sys
+# 3. io
+# 4. datetime
+# 5. requests
+# 6. dotenv
+# 
+# **Запрещённые модули**:
+# 1. Любые библиотеки, реализующие Telegram API в Python (*python-telegram-bot, Telethon, pyrogram, aiogram, telebot* и так далле...)
+# 2. Библиотеки, занимающиеся "перехватыванием" данных из `stdout` и `stderr` (*pytest-capturelog, contextlib, logging*  и так далле...)
+# 
+# 
+# 
+# Результат запуска кода ниже должен быть примерно такой:
+# 
+# ![image.png](attachment:620850d6-6407-4e00-8e43-5f563803d7a5.png)
+# 
+# ![image.png](attachment:65271777-1100-44a5-bdd2-bcd19a6f50a5.png)
+# 
+# ![image.png](attachment:e423686d-5666-4d81-8890-41c3e7b53e43.png)
+
+# In[24]:
+
+
+import os
+import sys
+import io
+import datetime
+import requests
+import dotenv
+import time
+#import subprocess
+my_chat_id = "781351382"
+dotenv.load_dotenv()
+
+
+# In[25]:
+
+
+def telegram_logger(chat_id):
+    def decorator(func):
+        def obyortka(*args, **kwargs):   # Поменять имя в перспективе
+            try:
+                url = f"https://api.telegram.org/bot{os.getenv('TG_API_TOKEN')}/sendMessage"
+                
+                time_start = datetime.datetime.now()
+                func(*args, **kwargs)
+                time_stop = datetime.datetime.now()
+                delta = time_stop - time_start
+                if delta.days >= 1:  
+                    delta = str(delta + datetime.timedelta(days=2))[:-7] 
+                else:
+                    delta = str(delta)
+        
+                positive = f"Function <code>{func.__name__}</code> succesfully finished in: {delta}"
+                
+                requests.post(url, data={"chat_id": chat_id,
+                                    "parse_mode": "HTML",
+                                    "text": positive})
+            except BaseException as error:
+                negative = f"Function <code>{func.__name__}</code> failed with an exception:\n<code>{error.__repr__()}</code>"
```

Comment:
Если есть возможность не писать дандер, то лучше не писать дандер
```suggestion
                negative = f"Function <code>{func.__name__}</code> failed with an exception:\n<code>{repr(error)}</code>"
```